### PR TITLE
Run tests and integration tests on Mac & Python3.7

### PR DESCRIPTION
**Pull request by crcrpar**
## Motivation

Run tests including integration on Mac every time the master branch gets updated on Python3.7.

The limitation of trigger and python version is for the sake of speed.

### CODE REVIEW

**Comment by crcrpar** [comment]
It appears to succeed 🎉 
image

File: .github/workflows/mac-tests.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    branches:
+      - main
+    paths-ignore:
+      - '**.md'
+      - 'docs/**'
+      - 'examples/**'
+      - 'tutorial/**'
+      - 'pyproject.toml'
+  # TODO(crcrpar): Remove once confirmed.
+  pull_request: {}
```
Comment (crcrpar) [inline .github/workflows/mac-tests.yml:17]:
```suggestion
```

**Comment by toshihikoyanase** [comment]
@hvy I think you talked with the author about this change, so could you review this PR?

File: .github/workflows/mac-tests.yml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+jobs:
+  tests:
+    runs-on: macos-latest
+
```
Comment (toshihikoyanase) [inline .github/workflows/mac-tests.yml:22]: This job will be executed in the forked repository. How about restricting it like the other jobs?

```suggestion
    runs-on: macos-latest

    # Not intended for forks.
    if: github.repository == 'optuna/optuna'
```

File: .github/workflows/mac-tests.yml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        pip install scikit-optimize
+        pip install cma
+
+    - name: Tests
+      run: |
+        pytest tests --ignore tests/integration_tests
+
+  tests-integration:
+    runs-on: macos-latest
+
```
Comment (toshihikoyanase) [inline .github/workflows/mac-tests.yml:63]: ditto.
```suggestion
    runs-on: macos-latest

    # Not intended for forks.
    if: github.repository == 'optuna/optuna'
```

File: .github/workflows/mac-tests.yml [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    - name: Tests
+      run: |
+        # pytest cannot collect allennlp tests with allennlp==1.0.0.
+        pytest -s tests/integration_tests
+
+    - name: Tests MPI
+      run: |
+        mpirun -n 2 -- pytest tests/integration_tests/test_chainermn.py
+      env:
```
Comment (toshihikoyanase) [inline .github/workflows/mac-tests.yml:109]: I think `test_pytorch_distributed.py` will also work on macOS.
```suggestion
        mpirun -n 2 -- pytest tests/integration_tests/test_chainermn.py
        mpirun -n 2 -- pytest tests/integration_tests/test_pytorch_distributed.py
      env:
```

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your PR. I confirmed that it worked on GitHub Actions.
It basically looks good to me except for small comments.

File: .github/workflows/mac-tests.yml (hunk 2 above)
Comment (crcrpar) [reply .github/workflows/mac-tests.yml:22]: okay, but I'd like CI to run on forked repositories.

File: .github/workflows/mac-tests.yml [hunk 5]
```diff
@@ -0,0 +1,115 @@
+# Run tests and integration tests on Mac, which are triggered by each master push.
+# Currently, Python3.7 is only used as an environment.
+# This is mainly for the sake of speed.
+name: mac-tests
+
+on:
+  push:
+    branches:
+      - main
```
Comment (crcrpar) [inline .github/workflows/mac-tests.yml:9]:
```suggestion
      - master
```

File: .github/workflows/mac-tests.yml [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    paths-ignore:
+      - '**.md'
+      - 'docs/**'
+      - 'examples/**'
+      - 'tutorial/**'
+      - 'pyproject.toml'
```
Comment (crcrpar) [inline .github/workflows/mac-tests.yml:15]:
```suggestion
```

**Comment by crcrpar** [comment]
Check whether we need openblas for MXnet 1.8. See #2508

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your update. LGTM.

**Pull request merged**